src/base64.h

```diff
@@ -129,12 +129,6 @@ static size_t base64_encode(const char* src,
                             size_t slen,
                             char* dst,
                             size_t dlen) {
-  // We know how much we'll write, just make sure that there's space.
-  CHECK(dlen >= base64_encoded_size(slen) &&
-        "not enough space provided for base64 encode");
-
-  dlen = base64_encoded_size(slen);
```

Comment:
@bnoordhuis do you mean 
`dlen = base64_encoded_size(slen);` is wrong change or the whole deletion is wrong including `CHECK` macro? 

I'm just curious how `dlen` that passed to the function can be different from returned `dlen`. 

returned `dlen` is` base64_encoded_size(slen)`.
passed `dlen` is the same `base64_encoded_size(slen)` according to [string_bytes.cc](https://github.com/nodejs/node/blob/master/src/string_bytes.cc#L735).

It would be great and really helpful if you or someone else can help me to understand it 🙏